src/v3/src/transformer/oktaVerify/transformOktaVerifyEnrollPoll.ts

```diff
@@ -74,20 +88,57 @@ export const transformOktaVerifyEnrollPoll: IdxStepTransformer = ({ transaction,
     // @ts-ignore OKTA-496373 - missing props from interface
     selectedChannel, phoneNumber = '', email = '', devicebootstrap: deviceBootstrap, samedevice: sameDevice,
   } = authenticator.contextualData;
+
   const enrolledDevices = deviceBootstrap?.enrolledDevices || [];
   const enrolledDevice = Array.isArray(enrolledDevices) ? enrolledDevices?.[0] : enrolledDevices;
   const qrCodeHref = authenticator.contextualData?.qrcode?.href;
+  let channelType = selectedChannel;
+  let deviceMap = [];
```

Comment:
Why is this initialized as `[]` instead of `{}`? Seems like `deviceMap` ends up getting set to either the `sameDevice` or `deviceBootstrap` object?

Reply:
otherwise I'm getting errors like `platform key does not exist` on the line 106. `deviceMap` is set to  `sameDevice` or `deviceBootstrap` if those exist and contain `setupOVUrl` property in any of them

Reply:
More than likely it is typescript complaining because there are no defined types in the object. 

You can do something like this to properly define the types and avoid initializing it as an array then changing it to an object: 

```
let deviceMap: <{ setupOVUrl?: string; platform?: string; platformLC?: string; isDesktop?: boolean; ... }> = {};
```

Reply:
thanks @glenfannin-okta ! please see https://github.com/okta/okta-signin-widget/pull/3526/commits/1fc893bc15a62aaebc731bb95533a243677189b3